packages/mrm-task-lint-staged/index.js

```diff
@@ -179,22 +180,48 @@ module.exports = function task({ lintStagedRules }) {
 		.unset('scripts.precommit')
 		// Remove husky 4 config
 		.unset('husky')
+		// Remove simple-git-hooks
+		.unset('simple-git-hooks')
 		// Add new config
 		.merge({
-			'simple-git-hooks': {
-				'pre-commit': 'npx lint-staged',
-			},
 			'lint-staged': rules,
-		})
-		.save();
+		});
+
+	if (isYarn2) {
+		// Yarn 2 doesn't support `prepare` lifecycle yet
+		// https://yarnpkg.com/advanced/lifecycle-scripts
+		pkg.appendScript('postinstall', 'husky install');
+		if (!pkg.get('private')) {
+			// In case package isn't private, pinst ensures that postinstall
+			// is disabled on publish
+			pkg
+				.appendScript('prepublishOnly', 'pinst --disable')
+				.appendScript('postpublish', 'pinst --enable');
+			packages.pinst = '>=2';
+		}
+	} else {
+		// npm, Yarn 1, pnpm
+		pkg.appendScript('prepare', 'husky install');
+	}
 
-	uninstall('husky');
+	pkg.save();
+
+	uninstall('simple-git-hooks');
 	// Install dependencies
 	install(packages);
+	// Install husky
+	husky.install();
+	// Set lint-staged config
+	husky.add('.husky/pre-commit', 'npx lint-staged');
 };
 
 module.exports.description = 'Adds lint-staged';
 module.exports.parameters = {
+	isYarn2: {
```

Comment:
Could we use [this method](https://github.com/sapegin/mrm/blob/481a3161bc9c1a778a27b73cd746f4a4d756a41d/packages/mrm-core/src/npm.js#L275-L280) to autodetect Yarn 2? Or Yarn 2 and Yarn Berry aren't the same?

Reply:
I think we could use it. Yarn Berry and Yarn 2 are the same.

Reply:
Can I export it in https://github.com/sapegin/mrm/blob/481a3161bc9c1a778a27b73cd746f4a4d756a41d/packages/mrm-core/src/npm.js#L278-L285

Reply:
Updated, should be good now :)